File: src/main/webapp/js/service/unit-service.js
mainApp.service('unitService', ['$http', '$q', 'GlobalHelper',  function($http, $q, GlobalHelper){
	// Return public API.
    return({
    	getAllWorkScopes:		getAllWorkScopes,
    	getUnitOfMeasurementList: getUnitOfMeasurementList,
    	getAppraisalPerformanceGroupMap: getAppraisalPerformanceGroupMap,
    	getSCStatusCodeMap: getSCStatusCodeMap
    });
   
    function getAllWorkScopes(){
    	var request = $http.get('service/unit/getAllWorkScopes');
    	return( request.then( GlobalHelper.handleSuccess, GlobalHelper.handleError ) );
    }
    
    function getUnitOfMeasurementList(){
    	var request = $http.get('service/unit/getUnitOfMeasurementList');
    	return( request.then( GlobalHelper.handleSuccess, GlobalHelper.handleError ) );
    }

    function getAppraisalPerformanceGroupMap(){
    	var request = $http.get('service/unit/getAppraisalPerformanceGroupMap');
    	return( request.then( GlobalHelper.handleSuccess, GlobalHelper.handleError ) );
    }

    function getSCStatusCodeMap(){
    	var request = $http.get('service/unit/getSCStatusCodeMap');
    	return( request.then( GlobalHelper.handleSuccess, GlobalHelper.handleError ) );
    }

}]);

File: src/main/java/com/gammon/qs/service/UnitService.java
package com.gammon.qs.service;

import java.util.List;
import java.util.Map;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import com.gammon.qs.application.exception.DatabaseOperationException;
import com.gammon.qs.dao.UnitWSDao;
import com.gammon.qs.domain.UnitOfMeasurement;
import com.gammon.qs.wrapper.UDC;
@Service
@Transactional(rollbackFor = Exception.class, value = "transactionManager")
public class UnitService {
	@Autowired
	private UnitWSDao unitDao;

	public List<UnitOfMeasurement> getUnitOfMeasurementList() throws Exception {
		return unitDao.getUnitOfMeasurementList();
	}

	public List<UDC> getAllWorkScopes() throws DatabaseOperationException {
		return unitDao.obtainWorkScopeList();
	}

	public UDC obtainWorkScope(String workScopeCode) throws DatabaseOperationException {
		return unitDao.obtainWorkScope(workScopeCode);
	}

	public Map<String, String> getAppraisalPerformanceGroupMap() {
		return unitDao.getAppraisalPerformanceGroupMap();
	}
	
	public Map<String, String> getSCStatusCodeMap() {
		return unitDao.getSCStatusCodeMap();
	}
}
